xla/service/gpu/model/indexing_analysis.cc

```diff
@@ -684,7 +684,7 @@ std::vector<int64_t> ToTransposeDimensions(const Layout& l) {
   return out;
 }
 
-llvm::SmallVector<AffineExpr, 4> DelinearizeInBoundsIndex(
+llvm::SmallVector<AffineExpr, 6> DelinearizeInBoundsIndex(
```

Comment:
The number of inlined elements is [decided based on the size of `T`](https://github.com/llvm/llvm-project/blob/0d6ed8399f35cc8b23a8e6d094379e3d0dd71c45/llvm/include/llvm/ADT/SmallVector.h#L1207) (= `AffineExpr` here), so seems to me a more robust fix would be to let the compiler calculate that number similar to the declaration of `result` in the function body:
```suggestion
llvm::SmallVector<AffineExpr> DelinearizeInBoundsIndex(
```
Otherwise if the size of `AffineExpr` changes in the future, the bug will resurface.